[Cocoa] Replace uses of CMNotificationCenter with CFNotificationCenter

aestes opened the pull request:
#### 5903034893af12971f9b3ee3426abaaa710d5af1
<pre>
[Cocoa] Replace uses of CMNotificationCenter with CFNotificationCenter
<a href="https://bugs.webkit.org/show_bug.cgi?id=260415">https://bugs.webkit.org/show_bug.cgi?id=260415</a>
rdar://112505255

Reviewed by Eric Carlson.

CMNotificationCenter is SPI that wraps CFNotificationCenter API on Cocoa platforms. As such, this PR
replaces its usage in WebKit with CFNotificationCenter. While here, this PR removes the code that
observed kCMSampleBufferConsumerNotification_BufferConsumed since
-[AVSampleBufferDisplayLayer prerollDecodeWithCompletionHandler:] is available on all OS versions
supported on main.

* Source/WebCore/PAL/pal/cf/CoreMediaSoftLink.cpp:
* Source/WebCore/PAL/pal/cf/CoreMediaSoftLink.h:
* Source/WebCore/PAL/pal/cocoa/MediaToolboxSoftLink.h:
* Source/WebCore/PAL/pal/spi/cf/CoreMediaSPI.h:
* Source/WebCore/PAL/pal/spi/cocoa/AVFoundationSPI.h:
* Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateMediaSourceAVFObjC.mm:
(WebCore::timebaseEffectiveRateChangedCallback):
(WebCore::EffectiveRateChangedListener::stop):
(WebCore::EffectiveRateChangedListener::EffectiveRateChangedListener):
(WebCore::CMTimebaseEffectiveRateChangedCallback): Deleted.
* Source/WebCore/platform/graphics/avfoundation/objc/SourceBufferPrivateAVFObjC.mm:
(WebCore::m_logIdentifier):
(WebCore::SourceBufferPrivateAVFObjC::~SourceBufferPrivateAVFObjC):
(WebCore::SourceBufferPrivateAVFObjC::enqueueSample):
(WebCore::bufferWasConsumedCallback): Deleted.
* Source/WebCore/platform/graphics/cocoa/MediaPlayerPrivateWebM.mm:
* Source/WebKit/UIProcess/Cocoa/WebProcessPoolCocoa.mm:
(WebKit::webProcessPoolHighDynamicRangeDidChangeCallback):
(WebKit::WebProcessPool::registerHighDynamicRangeChangeCallback):

Canonical link: <a href="https://commits.webkit.org/267078@main">https://commits.webkit.org/267078@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ff222aff297138cb405aa3d6af67425093b4d98b

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/15606 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/15912 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/16280 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/17362 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/14641 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/15788 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/18861 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/16010 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/17179 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/15791 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/16254 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/13280 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/18107 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/13518 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/14082 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/21009 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/14536 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/14249 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/17515 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/14837 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/12577 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/14093 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/3736 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/18455 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/14657 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

eric-carlson commented on Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateMediaSourceAVFObjC.mm:
```diff
@@ -125,23 +125,32 @@ void effectiveRateChanged()
+    auto notification = kCMTimebaseNotification_EffectiveRateChanged;
+    auto object = timebase;
+
+    CFNotificationCenterRemoveObserver(center, observer, notification, object);
```
Nit: Are all of the local variables necessary?

aestes replied:
Nope

eric-carlson commented on Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateMediaSourceAVFObjC.mm:
```diff
@@ -125,23 +125,32 @@ void effectiveRateChanged()
+    auto object = timebase;
+    auto suspensionBehavior = static_cast<CFNotificationSuspensionBehavior>(0);
+
+    CFNotificationCenterAddObserver(center, observer, callback, notification, object, suspensionBehavior);
```
Ditto.

eric-carlson commented on Source/WebKit/UIProcess/Cocoa/WebProcessPoolCocoa.mm:
```diff
@@ -1162,14 +1162,13 @@ static void webProcessPoolHighDynamicRangeDidChangeCallback(CMNotificationCenter
+        auto callback = webProcessPoolHighDynamicRangeDidChangeCallback;
+        auto notification = kMTSupportNotification_ShouldPlayHDRVideoChanged;
+        auto object = MT_GetShouldPlayHDRVideoNotificationSingleton();
+        auto suspensionBehavior = static_cast<CFNotificationSuspensionBehavior>(0);
```
Ditto

aestes committed:
[Cocoa] Replace uses of CMNotificationCenter with CFNotificationCenter
https://bugs.webkit.org/show_bug.cgi?id=260415
rdar://112505255

Reviewed by Eric Carlson.

CMNotificationCenter is SPI that wraps CFNotificationCenter API on Cocoa platforms. As such, this PR
replaces its usage in WebKit with CFNotificationCenter. While here, this PR removes the code that
observed kCMSampleBufferConsumerNotification_BufferConsumed since
-[AVSampleBufferDisplayLayer prerollDecodeWithCompletionHandler:] is available on all OS versions
supported on main.

* Source/WebCore/PAL/pal/cf/CoreMediaSoftLink.cpp:
* Source/WebCore/PAL/pal/cf/CoreMediaSoftLink.h:
* Source/WebCore/PAL/pal/cocoa/MediaToolboxSoftLink.h:
* Source/WebCore/PAL/pal/spi/cf/CoreMediaSPI.h:
* Source/WebCore/PAL/pal/spi/cocoa/AVFoundationSPI.h:
* Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateMediaSourceAVFObjC.mm:
(WebCore::timebaseEffectiveRateChangedCallback):
(WebCore::EffectiveRateChangedListener::stop):
(WebCore::EffectiveRateChangedListener::EffectiveRateChangedListener):
(WebCore::CMTimebaseEffectiveRateChangedCallback): Deleted.
* Source/WebCore/platform/graphics/avfoundation/objc/SourceBufferPrivateAVFObjC.mm:
(WebCore::m_logIdentifier):
(WebCore::SourceBufferPrivateAVFObjC::~SourceBufferPrivateAVFObjC):
(WebCore::SourceBufferPrivateAVFObjC::enqueueSample):
(WebCore::bufferWasConsumedCallback): Deleted.
* Source/WebCore/platform/graphics/cocoa/MediaPlayerPrivateWebM.mm:
* Source/WebKit/UIProcess/Cocoa/WebProcessPoolCocoa.mm:
(WebKit::webProcessPoolHighDynamicRangeDidChangeCallback):
(WebKit::WebProcessPool::registerHighDynamicRangeChangeCallback):

Canonical link: https://commits.webkit.org/267078@main
